removing dead code in trackerTPC

carlosperezlara opened the pull request.

mccumbermike commented:
Hi Carlos, I see some small changes to the tracking performance:
![carlos](https://cloud.githubusercontent.com/assets/12105552/16857219/b95d2534-49db-11e6-9302-fb0e6b5b65de.png)
Let me know if these are intentional or not. To double check I'm not doing the cross check incorrectly, I also ran a check of the latest new build but it is still the same:
![newbuild](https://cloud.githubusercontent.com/assets/12105552/16857237/d2b6218e-49db-11e6-84cd-528fef0d881f.png)

carlosperezlara commented:
Hi Mike,

No, this improvement was not intentional. I removed two thirds of code (namely 2.1k lines); lines that were not called during my tests. I only tried soft pions, though.

This suggests two things to me:

1) Some of these lines were actually called in some particular scenarios and I made a mistake thinking they were not.

2) The base class alternative for one of the functions I removed may provide a better tracking solution for the TPC case (sTPCTracker inherits from HelixHough and overwrites some methods…)

In any case, I recommend not to merge it just yet. I would like to understand the origin of those changes first: that might be a better payoff at the end.

Regards,



> On Jul 14, 2016, at 5:59 PM, Michael P. McCumber <notifications@github.com> wrote:
> 
> Hi Carlos, I see some small changes to the tracking performance:
>  <https://cloud.githubusercontent.com/assets/12105552/16857219/b95d2534-49db-11e6-9302-fb0e6b5b65de.png>
> Let me know if these are intentional or not. To double check I'm not doing the cross check incorrectly, I also ran a check of the latest new build but it is still the same:
>  <https://cloud.githubusercontent.com/assets/12105552/16857237/d2b6218e-49db-11e6-84cd-528fef0d881f.png>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/sPHENIX-Collaboration/coresoftware/pull/180#issuecomment-232805045>, or mute the thread <https://github.com/notifications/unsubscribe/ALAXaPQtIDgBcGIQI0YxBukU7qpEjU-wks5qVrErgaJpZM4JMqxZ>.
>

carlosperezlara commented:
following up on my comments during the Simulation meeting.

After letting breakrecursion do a selection on nlayers, we recover efficiency, but still the vertex residuals are a bit different.
Will keep investigating and reopen pull request when things are fully understood.

![daily](https://cloud.githubusercontent.com/assets/11540328/16959515/7c59cbfc-4db3-11e6-91e1-51a67453b82b.png)

carlosperezlara commented:
Comparison of this commit (solid) versus sPHENIX head (open)
![daily](https://cloud.githubusercontent.com/assets/11540328/17087554/0236f2ba-51db-11e6-9b56-4501eee13f8f.png)

Summary of functions removed in sPHENIXtrackerTPC::
> findHelicesParallel   /// designed to run in parallel core computers
> findHelicesParallelOneHelicity   /// designed to run in parallel core computers
> findHelicesParallelThread   /// designed to run in parallel core computers
> splitHitsParallelThread   /// designed to run in parallel core computers
> dcaToVertexXY   // not called
> fitTrack_3     // not called 
> initDummyHits     // not called
> findSeededTracks   /// designed find tracks by segments
> findSeededTracksbySegments   /// designed find tracks by segments
> findSeededTracksbySegments_run   /// designed find tracks by segments
> calculateKappaTangents   /// designed find tracks by segments
> projectToLayer   /// designed find tracks by segments
> findSeededTracksByProjection   /// designed find tracks by segments
> setRangeFromSeed   /// designed find tracks by segments
> kappaToPt   /// designed find tracks by segments
> ptToKappa   /// designed find tracks by segments

pinkenburg commented:
Hi Carlos,

I am a bit concerned about these kinds of global comparisons for changes 
which should have no effect whatsoever, especially when the before/after 
does show differences. What we've used in PHENIX and also for the 
calorimeters in sPHENIX is this ASCII dump which just prints all 
variables into a file and then you can run a diff to see if it's really 
identical (also useful to see if someone uses random number generators 
which use the system time as seed). The code is part of our coresoftware:

https://github.com/sPHENIX-Collaboration/coresoftware/tree/master/offline/packages/NodeDump

The tracks are not implemented yet but it's rather straightforward to do 
so. The 10 line macro to run this is:

https://github.com/sPHENIX-Collaboration/coresoftware/blob/master/offline/packages/NodeDump/macro/run_dump.C

This runs over a DST, but in principal one can run this as part of the 
reconstruction macro. (and don't forget to use a fixed random number 
seed in the G4 macro - that's my favorite mistake),

Chris

On 7/24/2016 8:13 PM, carlosperezlara wrote:
>
> Comparison of this commit (solid) versus sPHENIX head (open)
> daily 
> <https://cloud.githubusercontent.com/assets/11540328/17087554/0236f2ba-51db-11e6-9b56-4501eee13f8f.png>
>
> Summary of functions removed in sPHENIXtrackerTPC::
>
>     findHelicesParallel /// designed to run in parallel core computers
>     findHelicesParallelOneHelicity /// designed to run in parallel
>     core computers
>     findHelicesParallelThread /// designed to run in parallel core
>     computers
>     splitHitsParallelThread /// designed to run in parallel core computers
>     dcaToVertexXY // not called
>     fitTrack_3 // not called
>     initDummyHits // not called
>     findSeededTracks /// designed find tracks by segments
>     findSeededTracksbySegments /// designed find tracks by segments
>     findSeededTracksbySegments_run /// designed find tracks by segments
>     calculateKappaTangents /// designed find tracks by segments
>     projectToLayer /// designed find tracks by segments
>     findSeededTracksByProjection /// designed find tracks by segments
>     setRangeFromSeed /// designed find tracks by segments
>     kappaToPt /// designed find tracks by segments
>     ptToKappa /// designed find tracks by segments
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/sPHENIX-Collaboration/coresoftware/pull/180#issuecomment-234812297>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AG-irVsiEbVfGvczCT2vPEP7DcsV3XA5ks5qY_-PgaJpZM4JMqxZ>.
>

-- 
*************************************************************

Christopher H. Pinkenburg	;    pinkenburg@bnl.gov
				;    http://www.phenix.bnl.gov/~pinkenbu

Brookhaven National Laboratory	;    phone: (631) 344-5692
Physics Department Bldg 510 C	;    fax:   (631) 344-3253
Upton, NY 11973-5000

*************************************************************

mccumbermike commented:
This version is fine. 7 layer MAPS:
![maps](https://cloud.githubusercontent.com/assets/12105552/17198045/4d02dfb0-542f-11e6-972c-e7a57d65013d.png)

MAPS+TPC:
![tpc](https://cloud.githubusercontent.com/assets/12105552/17198051/55745264-542f-11e6-9cf8-b1d8a07fca19.png)

Merging...
